core/base/inc/TVersionCheck.h

```diff
@@ -39,7 +39,9 @@ class TVersionCheck {
 // visible in this TU.
 #ifndef R__CXXMODULES
 #ifndef __CINT__
+namespace ROOT {
 static TVersionCheck gVersionCheck(ROOT_VERSION_CODE);
```

Comment:
With this change we should not need the checks for the cxxmodules above. Could you try removing them?